checkov/secrets/scan_git_history.py

```diff
@@ -39,13 +42,16 @@ def __init__(self, root_folder: str, secrets: SecretsCollection,
         # in case we start from mid-history (git) we want to continue from where we've been
         self.history_store: GitHistorySecretStore = history_store or GitHistorySecretStore()
         self.raw_store: List[RawStore] = []
-        self.commits_diff: List[Commit] = []
+        self.repo: Repo
 
     def scan_history(self, last_commit_scanned: Optional[str] = '') -> bool:
         """return true if the scan finished without timeout"""
         timeout_class = ThreadingTimeout if platform.system() == 'Windows' else SignalTimeout
         # mark the scan to finish within the timeout
         with timeout_class(self.timeout) as to_ctx_mgr:
+            if not self.set_repo():
```

Comment:
why not in init ?

Reply:
tests :( other ides maybe to get it work with tests?

Reply:
other idea is to inject it between functions. not sure what is better